src/main/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/PeriodicShardSyncManager.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.amazonaws.services.kinesis.clientlibrary.lib.worker;
+
+import java.util.Objects;
+import java.util.Set;
+
+import com.amazonaws.services.kinesis.leases.impl.KinesisClientLease;
+import lombok.Getter;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * The top level orchestrator for coordinating the periodic shard sync related
+ * activities like polling for leader election, scheduling shard sync taks,
+ * responding to leader election events etc
+ */
+@Getter
+class PeriodicShardSyncManager {
+
+    private static final Log LOG = LogFactory.getLog(PeriodicShardSyncManager.class);
+
+    private final String workerId;
+
+    private final PeriodicShardSyncScheduler shardSyncScheduler;
+    private final ScheduledLeaseLeaderPoller leaderPoller;
```

Comment:
Let's eliminate this reference to `leaderPoller` in this class. The start/stop of the leader poller can happen inside start/stop methods of `shardSyncScheduler`. Doing this will further improve encapsulation.

Reply:
To me these appear to be different logical entities. One responsible for leader election and the other for scheduling the shard syncs

Reply:
Yes, and you can maintain logical separation of code in different classes. Using `leaderPoller` directly from `shardSyncScheduler` does not change that. Using it directly there simplifies code.

Reply:
Also, `shardSyncSchedule` already has a dependency on `leaderPoller`, so you wouldn't be introducing a new dependency either.